simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -412,71 +412,68 @@ class simd<double, simd_abi::avx2_fixed_size<4>> {
       const {
     return m_value;
   }
-  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION mask_type
-  operator<(simd const& other) const {
-    return mask_type(_mm256_cmp_pd(m_value, other.m_value, _CMP_LT_OS));
+  [[nodiscard]] KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd operator-() const
```

Comment:
What about `constexpr`?

Reply:
It's because these intrinsic functions are not `constexpr`.
Placing `constexpr` specifier will produce errors: `call to non-'constexpr' function '__m256d _mm256_sub_pd`.